shared/src/main/scala/squants/Quantity.scala

```diff
@@ -171,7 +171,7 @@ abstract class Quantity[A <: Quantity[A]] extends Serializable with Ordered[A] {
    * @return
    */
   override def equals(that: Any) = that match {
-    case x: Quantity[A] if x.dimension == dimension ⇒ value == x.to(unit)
+    case x: Quantity[_] if x.dimension == dimension ⇒ value == x.asInstanceOf[Quantity[A]].to(unit)
```

Comment:
Dotty prohibits this type assertion (Scala 2 should warn about this). I'm not entirely sure if my "fix" is correct.